File: src/main/java/com/mynimef/bot/IBot.java
package com.mynimef.bot;

import com.mynimef.bot.containers.BotMessage;
import com.mynimef.bot.containers.BotFile;

public interface IBot {

    Integer sendMessage(String chatId, BotMessage message);
    Integer sendMessage(String chatId, String text);
    void editMessage(String chatId, Integer messageId, String text);
    void editMessage(String chatId, Integer messageId, BotMessage message);
    void deleteMessage(String chatId, Integer messageId);
    void sendDoc(String chatId, BotFile file);
}

File: src/main/kotlin/com/mynimef/bot/IBot.kt
package com.mynimef.bot

import com.mynimef.bot.containers.BotFile
import com.mynimef.bot.containers.BotMessage

interface IBot {

    fun sendMessage(chatId: String, text: String): Int?

    fun sendMessage(chatId: String, message: BotMessage): Int?

    fun sendMessage(chatId: String, botFile: BotFile)

    fun editMessage(chatId: String, messageId: Int, message: BotMessage)

    fun editMessage(chatId: String, messageId: Int, text: String)

    fun deleteMessage(chatId: String, messageId: Int)

}
